File: eMRTDReader/sdk/src/main/java/com/example/emrtdreader/sdk/error/PassportReadException.java
package com.example.emrtdreader.sdk.error;

public class PassportReadException extends Exception {
    public PassportReadException(String message) { super(message); }
    public PassportReadException(String message, Throwable cause) { super(message, cause); }

    public static class TagNotIsoDep extends PassportReadException {
        public TagNotIsoDep() { super("Tag is not IsoDep"); }
    }

    public static class ReadFailed extends PassportReadException {
        public ReadFailed(Throwable cause) { super("Read failed", cause); }
        public ReadFailed(String message, Throwable cause) { super(message, cause); }
    }

    public static class AccessFailed extends PassportReadException {
        public AccessFailed(String message, Throwable cause) { super(message, cause); }
    }
}

File: eMRTDReader/app/src/main/java/com/example/emrtdreader/error/PassportReadException.kt
package com.example.emrtdreader.error

sealed class PassportReadException(message: String, cause: Throwable? = null) : Exception(message, cause) {
    class TagNotIsoDep : PassportReadException("Tag is not IsoDep compliant")
    class ReadFailed(cause: Throwable) : PassportReadException("Failed to read passport data", cause)
}
